src/os/bluestore/BlueFS.cc

```diff
@@ -833,16 +834,27 @@ int BlueFS::_check_new_allocations(const bluefs_fnode_t& fnode,
 }
 
 int BlueFS::_adjust_granularity(
-  __u8 id, uint64_t *offset, uint64_t *length, const char *op)
+  __u8 id, uint64_t *offset, uint64_t *length, bool alloc)
 {
+  const char *op = alloc ? "op_alloc_add" : "op_alloc_rm";
   auto oldo = *offset;
   auto oldl = *length;
   if (*offset & (alloc_size[id] - 1)) {
     *offset &= ~(alloc_size[id] - 1);
     *offset += alloc_size[id];
     if (*length > *offset - oldo) {
+      if (alloc) {
```

Comment:
You should process a tail left unused in the same way after doing
 *length &= ~(alloc_size[id] - 1);
below.